src/Microsoft.TemplateEngine.Edge/GlobalSettingsTemplatesSources/GlobalSettingsTemplatesSourcesProvider.cs

```diff
@@ -15,60 +17,64 @@ public partial class GlobalSettingsTemplatesSourcesProviderFactory
                 (GlobalSettingsTemplatesSourcesProviderFactory factory, IEngineEnvironmentSettings settings)
             {
+                _ = factory ?? throw new ArgumentNullException(nameof(factory));
+                _ = settings ?? throw new ArgumentNullException(nameof(settings));
+
                 Factory = factory;
                 PackagesFolder = Path.Combine(settings.Paths.TemplateEngineRootDir, "packages");
                 if (!settings.Host.FileSystem.DirectoryExists(PackagesFolder))
                 {
                     settings.Host.FileSystem.CreateDirectory(PackagesFolder);
                 }
-                this.settings = settings;
+
+                _environmentSettings = settings;
                 foreach (var installerFactory in settings.SettingsLoader.Components.OfType<IInstallerFactory>())
                 {
                     var installer = installerFactory.CreateInstaller(this, settings, PackagesFolder);
-                    installersByName[installerFactory.Name] = installer;
-                    installersByGuid[installerFactory.Id] = installer;
+                    _installersByName[installerFactory.Name] = installer;
+                    _installersByGuid[installerFactory.Id] = installer;
                 }
-                settings.SettingsLoader.GlobalSettings.SettingsChanged += SourcesChanged;
-            }
 
-            public ITemplatesSourcesProviderFactory Factory { get; }
+                ReloadCache();
+                settings.SettingsLoader.GlobalSettings.SettingsChanged += ReloadCache;
+            }
 
             public event Action SourcesChanged;
 
+            public ITemplatesSourcesProviderFactory Factory { get; }
+
             public Task<IReadOnlyList<ITemplatesSource>> GetAllSourcesAsync(CancellationToken cancellationToken)
             {
-                var list = new List<ITemplatesSource>();
-                var userSettings = settings.SettingsLoader.GlobalSettings;
-                foreach (var entry in userSettings.UserInstalledTemplatesSources)
+                List<ITemplatesSource> templatesSources = new List<ITemplatesSource>();
+                foreach (Dictionary<string, IManagedTemplatesSource> sourcesByInstaller in _templatesSources.Values)
                 {
-                    if (installersByGuid.TryGetValue(entry.InstallerId, out var installer))
-                    {
-                        if (installer.Deserialize(this, entry.MountPointUri, entry.Details) is IManagedTemplatesSource managedSource)
-                        {
-                            list.Add(managedSource);
-                        }
-                    }
+                    templatesSources.AddRange(sourcesByInstaller.Values);
                 }
-                return Task.FromResult<IReadOnlyList<ITemplatesSource>>(list);
+                templatesSources.AddRange(_notSupportedSources.Select(source => new TemplatesSource(this, source.MountPointUri, source.LastChangeTime)));
```

Comment:
`source.LastChangeTime` usage, I think we should either remove `LastChangeTime` or use it for all `GlobalSettingsTemplatesSourcesProvider` sources.